feat: ✨ add context support to line-range syntax (#3344)

cavanaug opened the pull request:
docs(long-help.txt): 📚 add examples for N::C and N:M:C context syntax
docs(clap_app.rs): 📚 update CLI help text with context syntax examples
feat(line_range.rs): ✨ implement N::C and N:M:C parsing and add tests

I decided to just take a stab at fixing the issue #3344 that I submitted to allow for providing a line number and how many lines of context to show.

keith-hall reviewed (commented):
Thanks, this seems reasonable to me, I like that you explained the use case in the issue 🙂 I wonder, would it be possible to add some integration tests do you think, please? 🙂

cavanaug commented:
I added lines to the multiline.txt file so I had more flexibility in testing various ranges and context.   I updated 2 of the tests that used this file previously so things are still appropriate.
